tests/integration/aval/aval-tests.yml

```diff
@@ -0,0 +1,140 @@
+build-aval-docker:
+  image: docker:dind
+  stage: build
+  rules:
+    - if: $CI_PIPELINE_SOURCE == "schedule"
+  variables:
+    IMAGE_NAME_AVAL: aval-docker
+    DOCKERFILE_NAME: tests/integration/aval/aval-docker.Dockerfile
+  before_script:
+    - docker login -u "${CI_REGISTRY_USER}" -p "${CI_REGISTRY_PASSWORD}" "${CI_REGISTRY}"
+    - docker login -u "${CI_DOCKER_HUB_PULL_USER}" -p "${CI_DOCKER_HUB_PULL_PASSWORD}"
+  script:
+    - ${B} docker build -f "${DOCKERFILE_NAME}" -t "${CI_REGISTRY_IMAGE}/${IMAGE_NAME_AVAL}:main" .
+    - ${B} docker push "${CI_REGISTRY_IMAGE}/${IMAGE_NAME_AVAL}:main"
+
+.aval-template:
+  stage: test
+  needs:
+    - build-aval-docker
+    - build-amd64
+  variables:
+    IMAGE_NAME: torizoncore-builder-amd64
+    IMAGE_NAME_AVAL: aval-docker
+  rules:
+    - if: $CI_PIPELINE_SOURCE == "schedule"
+  image: ${CI_REGISTRY_IMAGE}/${IMAGE_NAME_AVAL}:main
+  script:
+    # pull latest build of TorizonCore Builder
+    - echo -e "\e[0Ksection_start:$(date +%s):pull_eval_tcb_section\r\e[0KPull TorizonCore Builder to be evaluated"
+    - ${T} docker login -u "${CI_DOCKER_HUB_PULL_USER}" -p "${CI_DOCKER_HUB_PULL_PASSWORD}"
+    - ${T} docker login -u "${CI_REGISTRY_USER}" -p "${CI_REGISTRY_PASSWORD}" "${CI_REGISTRY}"
+    - ${T} docker pull "${CI_REGISTRY_IMAGE}/${IMAGE_NAME}:${GITLAB_DOCKERREGISTRY_SUFFIX_LATEST}"
+    - echo -e "\e[0Ksection_end:$(date +%s):pull_eval_tcb_section\r\e[0K"
+    - cd tests/integration && python /aval/main.py --delegation-config $DELEGATION_CONFIG
+      --run-before-on-host "./run_all.sh --device --machine $TCB_MACHINE --report --tcb-tags requires-device
+      --tcb-custom-image ${CI_REGISTRY_IMAGE}/${IMAGE_NAME}:${GITLAB_DOCKERREGISTRY_SUFFIX_LATEST}"
+      # To circumvent the mismatch between Aktualizr's database and ostree when an external entity manages ostree without
+      # Aktualizr, install a different version than the one Aktualizr thinks it's currently installed (the one before TCB
+      # deploy), which externally forces both Aktualizr and ostree to sync. In particular, if we're running a nightly, install
+      # a release build and vice-versa
+    - |
+      toggle_release_type() {
+        local current_release_type="$1"
+        if [ "$current_release_type" == "release" ]; then
+          echo "nightly"
+        elif [ "$current_release_type" == "nightly" ]; then
+          echo "release"
+        fi
+      }
+    - TARGET_BUILD_TYPE=$(toggle_release_type "$TARGET_BUILD_TYPE") python /aval/main.py --delegation-config $DELEGATION_CONFIG
+      "echo 'Updating to a different TOS version than the TCB tests version, to circumvent the mismatch between Aktualizr'\''s database and ostree when an external entity manages ostree without Aktualizr'"
+    - (! grep "^not ok" workdir/reports/*)
+
+# $AVAL_TEST_ALL can be defined on a Gitlab schedule to run tests on all devices
+aval-apalis-imx6q-scarthgap-release:
```

Comment:
Do we *need* to test all these boards? It could be best to simply choose the two most powerful 32/64-bits SoMS instead.

Reply:
By default, the tests will only run on the Apalis iMX6 and Verdin iMX8MP. If, for some reason, it is necessary to run the tests for all devices, the user should simply set AVAL_TEST_ALL=True.

I created jobs for all devices in case of some specific TCB tests need it.

Reply:
@lucasbernardestoradex Tricky question: what happens if you set `AVAL_TEST_ALL=False`?

Reply:
@rborn-tx Good point, the logic was "if AVAL_TEST_ALL is defined, run on all boards".
Actually, this was not what I wanted, so I changed to "if AVAL_TEST_ALL is defined as TRUE, run on all boards".